src/PaymentGateways/PayPalCommerce/PayPalCheckoutSdk/PayPalHttpClient.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Give\PaymentGateways\PayPalCommerce\PayPalCheckoutSdk;
+
+use Give\PaymentGateways\PayPalCommerce\Models\MerchantDetail;
+use PayPalCheckoutSdk\Core\FPTIInstrumentationInjector;
+use PayPalCheckoutSdk\Core\GzipInjector;
+use PayPalCheckoutSdk\Core\PayPalEnvironment;
+
+/**
+ * Class PayPalHttpClient.
+ *
+ * This class extends PayPalCheckoutSdk\Core\PayPalHttpClient class.
+ * PayPalHttpClient from php sdk force access token refresh on each http request.
+ * We register AuthorizationInjector class to inject access token in http request header.
+ * AuthorizationInjector clas refresh access token only if expired.
+ *
+ * @unreleased x.x.x
+ */
+class PayPalHttpClient extends \PayPalCheckoutSdk\Core\PayPalHttpClient
+{
+    /**
+     * Class constructor.
+     *
+     * @unreleased x.x.x
+     */
+    public function __construct(PayPalEnvironment $environment, $refreshToken = null)
+    {
+        parent::__construct($environment);
+        $this->authInjector = $this->getAuthorizationInjector($environment, $refreshToken);
+        $this->addInjector($this->authInjector);
+        $this->addInjector(new GzipInjector());
+        $this->addInjector(new FPTIInstrumentationInjector());
+    }
+
+    /**
+     * Returns AuthorizationInjector.
+     *
+     * @unreleased x.x.x
+     */
+    private function getAuthorizationInjector($environment, $refreshToken): AuthorizationInjector
+    {
+        $merchant = give(MerchantDetail::class);
+        $authorizationInjector = new AuthorizationInjector($this, $environment, $refreshToken);
+
+        // Set access token if exists.
+        if ($merchant->accessToken) {
```

Comment:
@JasonTheAdams To force refresh access token on every request for testing, you can either comment this `if` condition or set `false`.